mfa/push/src/main/kotlin/com/pingidentity/mfa/push/PushNotification.kt

```diff
@@ -0,0 +1,270 @@
+    val contextInfo: String? = null,
+    @Serializable(with = PushTypeSerializer::class)
+    val pushType: PushType,
+    @Serializable(with = DateSerializer::class)
+    val createdAt: Date = Date(),
+    @Serializable(with = NullableDateSerializer::class)
+    var respondedAt: Date? = null,
+    @Serializable(with = AdditionalDataSerializer::class)
+    val additionalData: Map<String, Any>? = null,
+    @EncodeDefault
+    var approved: Boolean = false,
+    @EncodeDefault
+    var pending: Boolean = true
+) {
+
+    /**
+     * String representation of the push notification type.
+     */
+    val type: String
+        get() = this.pushType.toString()
+
+    /**
+     * Checks if the user has responded to this notification.
+     *
+     * @return true if the notification has been approved or is not pending, false otherwise.
+     */
+    val responded: Boolean
+        get() = approved || !pending
+
+    /**
+     * Checks if this notification has expired.
+     *
+     * @return true if the notification has expired, false otherwise.
+     */
+    val expired: Boolean
+        get() {
+            val elapsedTimeSeconds = (System.currentTimeMillis() - createdAt.time) / 1000
+            return elapsedTimeSeconds > ttl
+        }
+
+    /**
+     * Mark this notification as approved.
+     */
+    fun markApproved() {
+        approved = true
+        pending = false
+        respondedAt = Date()
+    }
+
+    /**
+     * Mark this notification as denied.
+     */
+    fun markDenied() {
+        approved = false
+        pending = false
+        respondedAt = Date()
+    }
+
+    /**
+     * Get numbers used for push challenge
+     * @return The numbers as a read-only List<Int>.
+     *         Returns empty list if "numbersChallenge" is not available or cannot be parsed.
+     */
+    fun getNumbersChallenge(): List<Int> {
+        return this.numbersChallenge?.let { challengeString ->
+            try {
+                challengeString.split(",")
+                    .mapNotNull { it.trim().toIntOrNull() }
+            } catch (_: NumberFormatException) {
+                emptyList()
+            }
+        } ?: emptyList()
+    }
+
+    companion object {
+        /**
+         * Create a PushNotification from a JSON string.
+         *
+         * @param jsonString The JSON string.
+         * @return A PushNotification.
+         * @throws SerializationException if the JSON is invalid.
+         */
+        @JvmStatic
+        fun fromJson(jsonString: String): PushNotification {
+            val json = Json { 
+                ignoreUnknownKeys = true 
+                isLenient = true
+            }
+            return json.decodeFromString(jsonString)
+        }
+    }
+
+    /**
+     * Converts this notification to a JSON string representation.
+     *
+     * @return A JSON string representing this notification.
+     */
+    fun toJson(): String {
+        val json = Json { 
+            prettyPrint = false
```

Comment:
prettyPrint is required? Share the Json Instance.

Reply:
no, removed